### PR TITLE
Msisensor-ct

### PR DESCRIPTION
**Pull request by carmelinacharalambous**
### currently testing ###

### CODE REVIEW

File: tests/modules/msk/msisensor-ct/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { MSISENSOR-CT } from '../../../../modules/msk/msisensor-ct/scan/main.nf'
```
Comment (buehlere) [inline tests/modules/msk/msisensor-ct/main.nf]: Split these into two test for msi and scan. Correct include to correct name MSISENSOR-CT-MSI and MSISENSOR-CT-SCAN

File: tests/modules/msk/msisensor-ct/main.nf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+workflow test_msisensor_ct {
+
+    input = [
```
Comment (buehlere) [inline tests/modules/msk/msisensor-ct/main.nf]: input for empty test

File: tests/modules/msk/msisensor-ct/main.nf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        // customize test specific parameters here
+        "fasta",
+        [args: '--fa']
+
+    ]
+
+    MSISENSOR-CT ( input )
+}
+
+workflow test_msisensor_ct_data {
```
Comment (buehlere) [inline tests/modules/msk/msisensor-ct/main.nf]: input for test with data on juno. This test is similar to how you'd run an actual workflow using your module.

File: tests/modules/msk/msisensor-ct/nextflow.config [hunk 4]
```diff
@@ -0,0 +1,11 @@
+params {
+    enable_conda = false
+    fasta = "/home/buehlere/access_nextflow/getbasecountmultisample/chr22.fasta"
+    outdir = "output"
+}
+
+process {
```
Comment (buehlere) [inline tests/modules/msk/msisensor-ct/nextflow.config]: you can add optional parameters here with: `withName <module> { ext.args = '< option1 > < option2 >'} `

File: modules/msk/msisensor-ct/scan/main.nf [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        -d $fasta \\
+        -o ${prefix}.msisensor_scan.list \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        msisensor-ct: \$(msisensor-ct 2>&1 | sed -nE 's/Version:\\sv([0-9]\\.[0-9])/\\1/ p')
+    END_VERSIONS
+    """
+}
```
Comment (buehlere) [inline modules/msk/msisensor-ct/scan/main.nf]: add stub test here. See here: https://github.com/mskcc-omics-workflows/modules/blob/73b78e1dfa2121a120b8c681162525d428ef8bb7/modules/msk/gbcms/main.nf#L40-L47. We're faking the output files for the empty run of the module that we do here: https://github.com/mskcc-omics-workflows/modules/blob/73b78e1dfa2121a120b8c681162525d428ef8bb7/tests/modules/msk/gbcms/main.nf#L7-L24

File: modules/msk/msisensor-ct/scan/main.nf [hunk 6]
```diff
@@ -0,0 +1,33 @@
+process MSISENSOR_CT_SCAN {
+    tag "$meta.id"
+    label 'process_low'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        ghcr.io/msk-access/msisensor-ct:1.0 }"
+
+    input:
```
Comment (buehlere) [inline modules/msk/msisensor-ct/scan/main.nf:8]: you can add parameters here.

File: modules/msk/msisensor-ct/scan/main.nf [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+    output:
+    tuple val(meta), path("*.list"), emit: list
+    path "versions.yml"            , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
```
Comment (buehlere) [inline modules/msk/msisensor-ct/scan/main.nf:19]: edit the parameters here.

File: modules/msk/msisensor-ct/scan/main.nf [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    msisensor-ct \\
+        scan \\
+        -d $fasta \\
```
Comment (buehlere) [inline modules/msk/msisensor-ct/scan/main.nf:24]: add them to the command here.

File: tests/modules/msk/msisensor-ct/test.yml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+- name: msisensor-ct test_msisensor_ct
+  command: nextflow run ./tests/modules/msk/msisensor-ct -entry test_msisensor_ct -c ./tests/config/nextflow.config -c ./tests/modules/msk/msisensor-ct/nextflow.config -stub-run
```
Comment (buehlere) [inline tests/modules/msk/msisensor-ct/test.yml]: This is the command to actually run the stubtest. You can run this in your terminal to see if everything is working. Don't worry about the pytest for right now. Feel free to reach out if get an error.

If you want to run your 'real data test', you can change the `-entry` to another test name and remove `-stub-run`.

**Review by buehlere — CHANGES_REQUESTED** [review]
Feel free to reach out with any questions. I tried to summarize what we went over.

**Comment by rhshah** [comment]
@carmelinacharalambous  I know this is something we are not pursuing do you want to close this?

**Comment by carmelinacharalambous** [comment]
PR will be closed as we are no longer pursuing MSIsensor-ct

**Pull request closed without merging**